src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -367,6 +365,14 @@ public Participation findOneWithEagerResultsAndSubmissions(Long id) {
         log.debug("Request to get Participation : {}", id);
         Participation participation = findOneWithEagerResults(id);
         Hibernate.initialize(participation.getSubmissions());
+
+        participation.getSubmissions().forEach(submission -> {
+            Hibernate.initialize(submission.getResult()); // eagerly load the association
```

Comment:
This is again not very efficient. Assume a participation with 2*1000 submissions. In this case, we get 1000 additional hibernate calls. This also affects existing code that calls this method. Please try to find a more efficient method. We could e.g. discuss that submissions always eagerly load the corresponding result, if this is needed in most cases.
Then we still need to think how to improve the performance for assessors

Reply:
As far as I can tell, this is called only by my code, while it was already here it was unused.

I can try to think how to optimize it, I probably need only the latest submission actually, but how common is having more than 10 submissions for participation?

Reply:
You can simply write a method findOneWithEagerSubmissionsAndEagierResults()

Reply:
I was able to move to just one sql query, including retrieving the assessor